src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -60,16 +59,26 @@ public class RecordConverter {
 
     private final OpensearchSinkConnectorConfig config;
 
+    private final DocumentIDStrategy globalDocIdGenerator;
+    private final DocumentIDStrategy perTopicDocIdGenerator; // Used on topics present in TOPIC_KEY_IGNORE_CONFIG
+
     public RecordConverter(final Boolean useCompactMapEntries,
                            final RecordConverter.BehaviorOnNullValues behaviorOnNullValues) {
         this(null);
     }
 
     public RecordConverter(final OpensearchSinkConnectorConfig config) {
         this.config = config;
+        globalDocIdGenerator = (config != null && config.ignoreKey()) ? config.docIdStrategy()
+                                 : DocumentIDStrategy.RECORD_KEY; 
+        perTopicDocIdGenerator = (config != null) ? config.docIdStrategy() : DocumentIDStrategy.TOPIC_PARTITION_OFFSET;
+    }
+
+    private DocumentIDStrategy getDocIdStrategy(final String topic) {
+        return config.topicIgnoreKey().contains(topic) ? perTopicDocIdGenerator : globalDocIdGenerator;
     }
 
-    private String convertKey(final Schema keySchema, final Object key) {
+    public static String convertKey(final Schema keySchema, final Object key) {
```

Comment:
It is used only by `DocumentIDStrategy`, so it makes sense to move it there (and make it `private static`)